src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,346 @@
+---
+title: Asynchronous Dart code - futures, async, and await 
+description: A series of explanations and exercises for writing asynchronous code in Dart
+date: 2019-06-12
+tags: [ future, async, await, asynchronous, try, catch, error ]
+---
+
+## Introduction
+ 
+Welcome to the asynchronous Dart codelab! In this codelab you practice using
+Dart to run asynchronous code via the `async` and `await` keywords.
+
+In addition to explanations, this codelab includes embedded editors that show 
+you working code examples and coding exercises. You can use these editors to 
+test your knowledge by completing the exercises.
+
+Before you begin this codelab, you should:
+
+* Have programming experience in another language
+* Know basic syntax in Dart 
+* Have some experience writing asynchronous code in another language
+
+When you finish this codelab, you'll know the following: 
+
+* How to use the `async` keyword
+* How to use the `await` keyword
+* When to use `async` and `await` in your code
+* How your code executes when using the `async/await` keywords
+* How to handle errors from an asynchronous call using `try/catch` expressions 
+in async functions.
+* Where to find more information for the `async/await` material 
+covered in this codelab.
+
+
+## Why async matters
+
+Asynchronous operations let your program complete other work while waiting for 
+an operation to finish. Some common processes that require asynchronous 
+operations include: 
+
+* Fetching data over a network
+* Writing to a database
+* Reading data from a file
+
+To implement these features, you will need to learn the Dart syntax for
```

Comment:
To implement these features, you will need to learn the Dart syntax for
futures, async **(`async`)**, and await **(`await`)**.

-do you not need backticks around these words?

Reply:
I was wondering about this myself -- it felt awkward monospacing async/await but not futures:

...learn the Dart syntax for futures, `async`, and `await`

But then again, it's wrong to monospace futures:

...learn the Dart syntax for `futures`, `async`, and `await`

Since the word 'futures' isn't a language keyword or a classname. `Future` is of course a classname: 

...learn the Dart syntax for the `Future` class, and `async` and `await` keywords

But that sounds more verbose than it needs to be for a single sentence introduction.